ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -0,0 +1,185 @@
+package ch.epfl.ts.component.fetch
+
+import ch.epfl.ts.data.Quote
+import ch.epfl.ts.data.Currency
+import scala.util.parsing.combinator._
+import scala.io.Source
+import scala.concurrent.duration._
+import java.util.Date
+import java.util.Calendar
+import java.util.Timer
+import java.util.TimerTask
+import ch.epfl.ts.component.fetch.MarketNames._
+
+/**
+ * HistDataCSVFetcher class reads data from csv source and converts it to Quotes.
+ * For every Quote it has read from disk, it calls the function callback(q: Quote),
+ * simulating the past by waiting for a certain time t after each call. By default
+ * t is the original (historical) time difference between the quote that was last sent 
+ * and the next quote to be sent.
+ * 
+ * @param dataDir       A directory path containing data files the fetcher can read. Which files are
+ *                      actually read is determined by the start and end arguments (see below).
+ *                      The directory should contain substructures of the form <currency pair>/<xyz>.csv, e.g.: 
+ *                      EURCHF/DAT_NT_EURCHF_T_ASK_201304.csv,
+ *                      EURCHF/DAT_NT_EURCHF_T_BID_201304.csv,
+ *                      ...
+ *                      EURUSD/DAT_NT_EURUSD_T_BID_201305.csv, etc.
+ * @param currencyPair  The currency pair to be read from the data directory, e.g. "eurchf", "USDCHF", etc.
+ * @param start         From when to read. This can be any date, but will be reduced to its month. That means
+ *                      if start is set to 2013-04-24 14:34 the fetcher will start reading the first quote available
+ *                      in the data file for April 2013 (as if start was set to 2013-04-01 00:00).
+ * @param end           Until when to read. Behaves analogous to start, i.e. if end is set to 2013-06-24 14:34
+ *                      the fetcher will still read and send all data in June 2013, as if end was set to 2013-06-30 24:00 
+ * @param speed         The speed at which the fetcher will fetch quotes. Defaults to 1 (which means quotes are replayed
+ *                      as they were recorded). Can be set e.g. to 2 (time runs twice as fast) or 60 (one hour of historical
+ *                      quotes is sent in one minute), etc. Time steps are dT = int(1/speed * dT_historical), in milliseconds.
+ *                      Accordingly: be careful with high speeds, if e.g. speed=1000 the system can't make the difference
+ *                      between dT_historical=4200 and dT_historical=4900, they will be both sent after 4ms in the simulation.
+ *                      If we go even higher, dT will become < 1 and thus 0, which in theory would mean send all the quotes
+ *                      you read at the same time, but in practice would probably mess up things.
+ */
+
+class HistDataCSVFetcher(dataDir: String, currencyPair: String, 
```

Comment:
I would have liked to change the name to avoid abbreviations, but it's not too important.